thrust/detail/device_ptr.inl

```diff
@@ -41,13 +40,6 @@ template<typename T>
   return ptr;
 } // end device_pointer_cast()
 
-// output to ostream
-template<class E, class T, class Y>
-  std::basic_ostream<E, T> &operator<<(std::basic_ostream<E, T> &os, const device_ptr<Y> &p)
```

Comment:
I'd rather not introduce operator<< for pointer & device_ptr. Let's just fix the problems with reference for now.